core/services/document_service.py

```diff
@@ -401,6 +396,12 @@ async def ingest_text(
         )
         logger.debug(f"Created text document record with ID {doc.external_id}")
 
+        if settings.MODE == "cloud" and auth.user_id:
+            # Check limits before proceeding
+            from core.api import check_and_increment_limits
+            num_pages = int(len(content)/(4*630)) # 
```

Comment:
Can we make this a variable instead of magic number, eg. PAGE_SIZE or PAGE_THRESHOLD